routes/project/discount_code.rb

```diff
@@ -30,6 +30,13 @@ class Clover
         rescue Sequel::UniqueConstraintViolation
           flash["error"] = "Discount code has already been applied to this project."
         else
+          unless @project.billing_info
```

Comment:
everyone entering a discount code would be considered having a valid payment method, right? they'd have billing info and credit > 0. we should at least keep an eye on suspicious activity of projects without payment method but a coupon. 

```ruby
  def has_valid_payment_method?
    return true unless Config.stripe_secret_key
    !!billing_info&.payment_methods&.any? || (!!billing_info && credit > 0)
  end
  ```
  https://github.com/ubicloud/ubicloud/blob/6d568f373bb4a8b0ba75dc5c2ca73fcab1165a7e/model/project.rb#L40